backend/FwLite/MiniLcm/SyncHelpers/SenseSync.cs

```diff
@@ -0,0 +1,63 @@
+﻿using MiniLcm.Models;
+using SystemTextJsonPatch;
+
+namespace MiniLcm.SyncHelpers;
+
+public static class SenseSync
+{
+    public static async Task<int> Sync(Guid entryId,
+        Sense afterSense,
+        Sense beforeSense,
+        IMiniLcmApi api)
+    {
+        var updateObjectInput = await SenseDiffToUpdate(beforeSense, afterSense);
+        if (updateObjectInput is not null) await api.UpdateSense(entryId, beforeSense.Id, updateObjectInput);
+        var changes = await ExampleSentenceSync.Sync(entryId,
+            beforeSense.Id,
+            afterSense.ExampleSentences,
+            beforeSense.ExampleSentences,
+            api);
+        return changes + (updateObjectInput is null ? 0 : 1);
+    }
+
+    public static async Task<UpdateObjectInput<Sense>?> SenseDiffToUpdate(Sense beforeSense, Sense afterSense)
+    {
+        JsonPatchDocument<Sense> patchDocument = new();
+        patchDocument.Operations.AddRange(
+            MultiStringDiff.GetMultiStringDiff<Sense>(nameof(Sense.Gloss), beforeSense.Gloss, afterSense.Gloss));
+        patchDocument.Operations.AddRange(MultiStringDiff.GetMultiStringDiff<Sense>(nameof(Sense.Definition),
+            beforeSense.Definition,
+            afterSense.Definition));
+        if (beforeSense.PartOfSpeech != afterSense.PartOfSpeech)
```

Comment:
Having both `PartofSpeech` and `PartOfSpeechId` is code debt, right?
I feel like it was a quick hack somewhere along the line. Should there be an issue to clean that up?

Reply:
I'm not sure. The same could be said with how Semantic domains are stored, we could just store a list of guids, but instead we store the whole object, otherwise the client needs to fetch the list of domains for those guids to mean anything. So maybe the answer here is to use the Part of speech object on the sense?

Reply:
I see now. I thought `PartOfSpeech` was residue from LfClassic stuff.

It seems odd that we have Semantic domain objects, but not Part of speech objects. I suppose we need Semantic domain objects, because we have a list of them. So, we can live without PoS objects.

One advantage of a PoS object, is that we could make it immutable, which would make it harder for the two properties to get out of sync. To do that we'd probably have to start distinguishing between the "whole" object (i.e. with `DeletedAt` and `Predefined` etc.) and a "minimal" object that is added to the entity for an easy-access display value.

Reply:
yeah that's a good point, we could just use the object. That would also resolve the name missmatch where the name here is a string, but it's a multi text on the PoS

Reply:
let's do this here #1232